Makefile

```diff
@@ -149,20 +149,14 @@ PROTO_SRC_NAMES = $(basename $(notdir $(PROTO_SRCS)))
 PROTO_GO_OUTS = $(foreach name, $(PROTO_SRC_NAMES), go/vt/proto/$(name)/$(name).pb.go)
 
 # This rule rebuilds all the go files from the proto definitions for gRPC.
-proto: $(PROTO_GO_OUTS)
-
+proto: $(PROTO_SRCS)
 ifndef NOBANNER
 	echo $$(date): Compiling proto definitions
 endif
-
-# TODO(sougou): find a better way around this temp hack.
-VTTOP=$(VTROOT)/../../..
-$(PROTO_GO_OUTS): install_protoc-gen-go proto/*.proto
-	for name in $(PROTO_SRC_NAMES); do \
-		cd $(VTTOP)/src && \
-		$(VTROOT)/bin/protoc --go_out=plugins=grpc:. -Ivitess.io/vitess/proto vitess.io/vitess/proto/$${name}.proto && \
-		goimports -w $(VTROOT)/go/vt/proto/$${name}/$${name}.pb.go; \
-	done
+	docker build -t proto-builder -f docker/proto/Dockerfile .
+	docker run --rm --mount type=bind,src=$(shell pwd)/go/vt/proto/,dst=/go/vt/proto/ proto-builder
```

Comment:
This is likey why you have to sudo below. Instead of bind mounting just use `-v` to mount the directory and run the container as the current user.

Reply:
this `mount` seems to have the unfortunate effect of making chmod-ing directories to `777`

Reply:
Yeah, I'm going to push up a commit switching to the `-v` style